rfcs/Subworlds.md

```diff
@@ -0,0 +1,139 @@
+# Feature Name: `Subworlds`
+
+## Summary
+
+Currently, `bevy` contains a single _global_ `World` which holds all entities/components, which are operated on via `Queries` and `Systems`. However, there are uses for disjoint sets of entities that can operate independently from one another. This RFC proposes a concept of `Subworlds`, which are individually a distinct collection of entities and their components. In this model, `Queries` and `Systems` can operate on these `Subworlds` independently and concurrently. This provides the ability to encapsulate sets of entities into `Subworlds`, reducing some of the complexity and overhead that came with grouping entities in the _global_ `World` (via _marker-components_ for example). By excluding large sets of entities that live in other `Subworlds`, `Systems` and `Queries` can more efficiently perform their logic on a smaller, more concise set of components.  
+
+## Motivation 🏃
+
+The primary motivation for using `Subworlds` to allows systems and queries to operate on disjoint sets of entities and their components.
+
+## Guide-level explanation 📝
+
+### Terms & Definitions 📚
+
+- `Subworld`: This refers to the original definition of a `World` in bevy 0.5. (essentially a collection of components which belong to a set of entities). 
```

Comment:
I'm not sure I think "Subworld" is a useful term. What is it part of, as implied by sub-?

Would "multiple Worlds" capture the same idea in a clearer way?